cloud/wfm-rest-api/test/MongoDbRepositoryTest.ts

```diff
@@ -10,45 +13,77 @@ const testObj = { id: '1', name: 'test' };
 
 const testSubject = new MongoDbRepository<TestEntity>('testCollection');
 const testSubjectNoDb = new MongoDbRepository<TestEntity>('testCollection');
-const mock = function() { return db; };
-const db: any = {
-  collection: mock,
-  findOne: mock,
-  insertOne: mock,
-  find() {
-    return {
-      count(filter, cb) {
-        cb(10);
-      }
-    };
+const mockDataAccessFunction = function() { return Bluebird.resolve(testObj); };
+const mockCursor: any = {
+  count() {
+    return Bluebird.resolve(10);
+  },
+  sort() {
+    return this;
+  },
+  skip() {
+    return this;
+  },
+  limit() {
+    return this;
   },
-  updateOne: mock,
-  deleteOne: mock
+  toArray() {
+    return [testObj];
+  }
+};
+const db: any = {
+  collection: () => ({
+    findOne: mockDataAccessFunction,
+    insertOne: mockDataAccessFunction,
+    find() {
+      return mockCursor;
+    },
+    updateOne: mockDataAccessFunction,
+    deleteOne() {
+      return {
+        result: {
+          ok: 1
+        }
+      };
+    }
+  })
 };
 testSubject.setDb(db);
 
 describe('FeedHenry MongoDbRepository Tests', function() {
   describe('Test CRUD operations', function() {
+
     it('should create obj', function() {
-      assert.ok(testSubjectNoDb.create(testObj));
-      assert.ok(testSubject.create(testObj));
+      return Bluebird.all([
+        expect(testSubject.create(testObj)).to.eventually.equal(testObj),
+        expect(testSubjectNoDb.create(testObj)).to.be.rejected
+      ]);
     });
+
     it('should get obj', function() {
-      assert.ok(testSubjectNoDb.get('test'));
-      assert.ok(testSubject.get('test'));
+      return Bluebird.all([
+        expect(testSubject.get('1')).to.eventually.equal(testObj),
+        expect(testSubjectNoDb.get('1')).to.be.rejected
+      ]);
     });
     it('should list obj', function() {
       const page = { order: -1, page: 0, size: 10, sortField: 'id' };
-      assert.ok(testSubject.list({}, page));
-      assert.ok(testSubjectNoDb.list({}, page));
+      return Bluebird.all([
```

Comment:
Really cool change. Thanks for fixing this as previous version was terrible.
Returning promise for tests is a good pattern. I did now know that this was supported.